Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/AutoIngestCaseManager.java

```diff
@@ -49,21 +59,81 @@ synchronized static AutoIngestCaseManager getInstance() {
+        try {
+            init();
+        } catch (AutoIngestCaseManager.AutoIngestCaseManagerException ex) {
+            Logger.getLogger(AutoIngestCaseManager.class.getName()).log(Level.SEVERE, null, ex);
+            throw ex;
+        }
+    }
+    
+    /**
+     * Initialize the coordination service.
+     * 
+     * @throws org.sleuthkit.autopsy.experimental.autoingest.AutoIngestCaseManager.AutoIngestCaseManagerException 
+     */
+    private void init() throws AutoIngestCaseManager.AutoIngestCaseManagerException {
+        try {
+            coordinationService = CoordinationService.getInstance();
+        } catch (CoordinationServiceException ex) {
+            throw new AutoIngestCaseManager.AutoIngestCaseManagerException(ex.getMessage(), ex);
+        }
     }
 
-    /*
+    /**
      * Gets a list of the cases in the top level case folder used by auto
      * ingest.
      */
+    @Messages({
+        "AutoIngestCaseManager.CasePathsError=An error occurred while trying to retrieve the list of case paths."
+    })
     List<AutoIngestCase> getCases() {
         List<AutoIngestCase> cases = new ArrayList<>();
-        List<Path> caseFolders = PathUtils.findCaseFolders(Paths.get(AutoIngestUserPreferences.getAutoModeResultsFolder()));
-        for (Path caseFolderPath : caseFolders) {
-            cases.add(new AutoIngestCase(caseFolderPath));
+        List<Path> casePathList;
+        try {
+            casePathList = getCasePaths();
+            for (Path casePath : casePathList) {
+                cases.add(new AutoIngestCase(casePath));
+            }
+        } catch (AutoIngestCaseManagerException ex) {
+            String errorMessage = NbBundle.getMessage(AutoIngestCaseManager.class, "AutoIngestCaseManager.CasePathsError");
+            LOGGER.log(Level.SEVERE, errorMessage, ex);
+            MessageNotifyUtil.Message.error(errorMessage);
         }
         return cases;
     }
+    
+    /**
+     * Retrieve all of the case nodes and filter for only those that represent
+     * case paths.
+     * 
+     * @return List of case paths.
+     */
+    private List<Path> getCasePaths() throws AutoIngestCaseManagerException {
+        try {
+            List<String> nodeList = coordinationService.getNodeList(CoordinationService.CategoryNode.CASES);
+            List<Path> casePathList = new ArrayList<Path>(0);
+            for (String node : nodeList) {
+                if(node.indexOf('\\') >= 0 || node.indexOf('/') >= 0) {
+                    /*
+                     * This is not a case name lock (name specifies a path).
+                     */
+                    String nodeUpperCase = node.toUpperCase();
+                    if(!nodeUpperCase.endsWith("_RESOURCES") && !nodeUpperCase.endsWith("AUTO_INGEST_LOG.TXT")) {
+                        /*
+                         * This is not a case resource lock. Collect the path.
+                         */
```

Comment:
For completeness, the comment should probably also mention that it is not a case auto ingest log lock node, either.